File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String,Integer> tmp = new HashMap<String,Integer>();
        String ans = "";
        for(String comp: completion){
            if(tmp.keySet().contains(comp)){
                tmp.replace(comp, tmp.get(comp)+1);
            }
            else{
                tmp.put(comp,1);
            }
        }
        Set<String> keyset = tmp.keySet();
        
        for(String part: participant){
            if(keyset.contains(part) && tmp.get(part) > 0){
                tmp.replace(part,tmp.get(part)-1);
            }
            else{
                ans = part;
            }
        }
        return ans;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    dic = defaultdict(int)
    for i in participant:
        dic[i] += 1
    
    for i in completion:
        dic[i] -= 1
        if dic[i] == 0:
            dic.pop(i)
    
    print(dic.keys())
    return list(dic.keys())[0]
        
    